src/auth/project-config-manager.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { App } from '../app';
+import { ProjectConfig, ProjectConfigServerResponse, UpdateProjectConfigRequest } from './project-config';
+import {
+  AuthRequestHandler,
+} from './auth-api-request';
+
+/**
+ * Defines the project config manager used to help manage project config related operations.
+ * This includes:
+ * <ul>
+ * <li>The ability to update and get project config.</li>
+ * <li>Getting a `TenantAwareAuth` instance for running Auth related operations
+ *     (user management, provider configuration management, token verification,
+ *     email link generation, etc) in the context of a specified tenant.</li>
+ * </ul>
+ */
+export class ProjectConfigManager {
+  private readonly authRequestHandler: AuthRequestHandler;
+  
+  /**
+   * Initializes a ProjectConfigManager instance for a specified FirebaseApp.
+   *
+   * @param app - The app for this ProjectConfigManager instance.
+   *
+   * @constructor
+   * @internal
+   */
+  constructor(app: App) {
+    this.authRequestHandler = new AuthRequestHandler(app);
+  }
+
+  /**
+   * Get the project configuration.
+   *
+   * @returns A promise fulfilled with the project configuration.
+   */
+  public getProjectConfig(): Promise<ProjectConfig> {
+    return this.authRequestHandler.getConfig()
+      .then((response: ProjectConfigServerResponse) => {
+        return new ProjectConfig(response);
+      })
+  }
+  /**
+   * Updates an existing project configuration.
+   *
+   * @param projectConfigOptions - The properties to update on the project.
+   *
+   * @returns A promise fulfilled with the update project config.
+   */
+  public updateProjectConfig(projectConfigOptions: UpdateProjectConfigRequest): Promise<ProjectConfig> {
+    return this.authRequestHandler.updateConfig(projectConfigOptions)
```

Comment:
Shall we update all the names to `updatePojectConfig`, config itself here is not informative enough

Reply:
Done